src/dom/position/dimensions.ts

```diff
@@ -0,0 +1,33 @@
+import { CachedElementVector } from '../cached-element-vector';
+import { Dimensions2dDom } from './dimensions-2d-dom';
+
+/**
+ * Caches element dimensions on a per-frame basis.
+ * Useful for measuring element distances as high level elements may be queried
+ * multiple times.
+ */
+export class Dimensions extends CachedElementVector<Dimensions2dDom> {
```

Comment:
So after further thought, I'm kind of more in the opinion that perhaps we don't need this level of caching in the library itself.  Sorry for going back and forth on this.

I think we want to uphold performance but trying to manage and caching at a PER frame level adds a lot of complexity and abstraction.  I think we want to keep things fairly simple here.

For sties and cases that may require this level of per frame cache optimization, these services should be within the project and not part of this library.

WTDY?

Reply:
i wonder if instead of having separate caches for each thing that may need caching, if we can instead have a single cache.

something like:

```javascript
const cache = new RafCache();  // gets reset on every frame
const windowHeight = cache.get('window.innerHeight', () => window.innerHeight);
```

then there wouldn't need to be any code to track "usages" of different types of values, and this code could be moved to the caller instead of having it within this library itself.

Reply:
i've been giving it more and more thought and i tend to agree with scott that maybe the cache-related code here may not make sense. it creates a level of interconnectedness across different parts of the code that, while may optimize things, makes the individual utility functions more difficult to maintain.

for example, with the change to `isDisplayNone()`, the function no longer simply tells you whether an element is set to `display:none`. if i understand the code correctly, it now fetches the style service (instantiates it if it isn't already instantiated), allocates a cache object, registers the cache object in its "usages" map, and subscribes to raf to clear its values on every frame.

for the most part, i think the library should focus on providing utilities that have simple, clear, easy-to-use apis. the extra classes that hyper optimize the dom are still useful, but maybe they should be moved into the project directories themselves, which would make it easier to maintain since those apis can be changed out rather easily without affecting any other clients who may also be using it.

Reply:
I will yank them. The uses/disposal could probably be done neater. I'm going to leave the benchmark here so we have a rough idea of potential gains if we ever need to revisit performance out of necessity: https://jsbench.me/eokiv6f8qy/1